services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/DefaultRulesSpec.groovy

```diff
@@ -129,6 +129,80 @@ class DefaultRulesSpec extends HealthCheckSpecification {
         }
     }
 
+    @Unroll
+    @Tags([TOPOLOGY_DEPENDENT, SMOKE_SWITCHES])
+    def "Able to install default multitable rule on an #sw.ofVersion \
+switch(#sw.dpId, install-action=#data.installRulesAction)"(Map data, Switch sw) {
+        setup: "Enabled multitable switch property on a switch(if it is not enabled yet)"
+        def initSwitchProperty = northbound.getSwitchProperties(sw.dpId)
+        def multitableIsEnabled = initSwitchProperty.multiTable
+        multitableIsEnabled ?: northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId).tap {
+            it.multiTable = true
+        })
+
+        and: "Delete all rules"
+        def defaultRules = northbound.getSwitchRules(sw.dpId).flowEntries
+        assert defaultRules*.cookie.sort() == sw.defaultCookies.sort()
+
+        northbound.deleteSwitchRules(sw.dpId, DeleteRulesAction.DROP_ALL)
+        Wrappers.wait(RULES_DELETION_TIME) { assert northbound.getSwitchRules(sw.dpId).flowEntries.empty }
+
+        when: "Install rules on the switch"
+        def installedRules = northbound.installSwitchRules(sw.dpId, data.installRulesAction)
+
+        then: "The corresponding rules are really installed"
+        installedRules.size() == 1
+
+        def expectedRules = defaultRules.findAll { it.cookie == data.cookie }
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            compareRules(northbound.getSwitchRules(sw.dpId).flowEntries, expectedRules)
+        }
+
+        and: "Install missing default rules and restore switch properties"
+        multitableIsEnabled ?: northbound.updateSwitchProperties(sw.dpId, initSwitchProperty)
```

Comment:
Instead of updating the switch table mode back and forth for every test iteration, you can use the `TestFixture` extension.
```
@TestFixture(setup="makeSwitchMultiTable", cleanup="revertSwitchMode")
def "myTestWithIterations"() {
...
}

def makeSwitchMultiTable() {
   //make desired switch multi table
}

def revertSwitchMode() {
   //revert the actions taken in 'makeSwitchMultiTable'
}

```

Reply:
Oh disregard this comment. Every iteration takes a different switch, so it makes no sense.